clients/admin-ui/src/features/datastore-connections/add-connection/forms/YamlEditorForm.tsx

```diff
@@ -91,17 +93,19 @@ const YamlEditorForm: React.FC<YamlEditorFormProps> = ({
 
   const handleConfirmation = () => {
     // Only need the confirmation if we are overwriting, which only happens when
-    // there is already data
-    if (data.length) {
+    // there are already datasets
+    if (allDatasets && allDatasets.length) {
       const value: string = (monacoRef.current as any).getValue();
-      // Check if the fides key that is in the editor is the same as the one that already exists
+      // Check if the fides key that is in the editor is the same as one that already exists
       // If so, then it is an overwrite and we should open the confirmation modal
-      if (data.some((d) => value.includes(`fides_key: ${d.fides_key}\n`))) {
+      if (
+        allDatasets.some((d) => value.includes(`fides_key: ${d.fides_key}\n`))
+      ) {
```

Comment:
The message in the warning modal may not always be technically correct: the ctl_dataset that we're about to upsert, is going to match whatever the `fides_key` is in the first line of the yaml file. 

So here, if the dataset was originally `my_other_mongo_test` and I have just pasted yaml that has `mongo_test`, it is `mongo_test` that is going to be either created or updated. 

<img width="1001" alt="Screen Shot 2023-01-12 at 11 12 43 AM" src="https://user-images.githubusercontent.com/9755598/212134377-269b1840-63f0-4822-b5b4-e959b3d79c4a.png">